Update README.md and add docs (#267)
scale-it/algo-builder: README.md

TODO: we are planning to provide a template for TS projects. [task](https://www.pivotaltracker.com/n/projects/2452320).

## Templates

> Using the `algob unbox-template` command, the developers can get a pre-built dApp project containing scripts to deploy assets and smart contracts with react.js interactive frontend. The templates heavily use [Algosigner](https://github.com/PureStake/algosigner) to securely sign and send transactions to the ledger. 

Reviewer: ```suggestion
Using the `algob unbox-template` command, the developers can get a pre-built dApp project containing scripts to deploy assets and smart contracts with react.js frontend. The templates use [AlgoSigner](https://github.com/PureStake/algosigner) to securely sign and send transactions to an Algorand Blockchain network. 
```

Author: Will make the changes.

> Using the `algob unbox-template` command, the developers can get a pre-built dApp project containing scripts to deploy assets and smart contracts with react.js interactive frontend. The templates heavily use [AlgoSigner](https://github.com/PureStake/algosigner) to securely sign and send transactions to an Algorand Blockchain Network. 

Reviewer 2: maybe we can first explain what user will get and then how i.e `algob unbox.....`. just a thought

Author: @amityadav0  this doc is just an overview of templates and `algob unbox-template`, The real documentation can be found in the repo itself, i.e., `algo-builder-templates`. I have provided the link to it in the starting itself.

Reviewer 2: what i meant here, our heading is `templates` so instead of starting with `algob unbox-templates` , we can first explain in a line what do we mean here and then the command. or this one is fine too.

Author: Okay. got it. I have made the apt changes.

> Using the `algob unbox-template` command, the developers can get a pre-built dApp project containing scripts to deploy assets and smart contracts with react.js interactive frontend. The templates heavily use [AlgoSigner](https://github.com/PureStake/algosigner) to securely sign and send transactions to an Algorand Blockchain Network. 

Reviewer 2: ```suggestion
Using the `algob unbox-template` command, the developers can get a pre-built dApp project containing scripts to deploy assets and smart contracts with react.js interactive frontend. The templates use [AlgoSigner](https://github.com/PureStake/algosigner) to securely sign and send transactions to an Algorand Blockchain Network. 
```

> The detailed description about the templates can be found [here](https://github.com/scale-it/algo-builder-templates).

Reviewer 3: ```suggestion
Detailed description about the templates can be found [here](https://github.com/scale-it/algo-builder-templates#algo-builder-templates).
```

Author: done
